### ISSUE TITLE
NPE when running scalac

### ISSUE THREAD

**Issue by thomaswue** [graalvm/graal#241]
See output and diagnostics zip at https://gist.github.com/retronym/0801d861ab914eb45f547f20dab86bf4

**Comment by retronym** [comment]
s/javac/scalac :)

**Comment by retronym** [comment]
I believe this is already fixed in 59bf33d83d8060692a086bd5563831da8e101a28.

**Comment by retronym** [comment]
I wasn't able to build master locally to test it out.

```
⚡ (export JAVA_HOME=$JAVA_HOME_18LABS; PATH=$JAVA_HOME_18LABS/bin:$PATH; mx build; mx vm)
...
Archiving TRUFFLE_TEST_NATIVE... [dependency com.oracle.truffle.nfi.test.native updated]
Archiving TRUFFLE_TEST... [dependency com.oracle.truffle.api.test updated]
Usage: java [-options] class [args...]
           (to execute a class)
   or  java [-options] -jar jarfile [args...]
           (to execute a jar file)
...
⚡ echo $?
1
```

**Comment by retronym** [comment]
I seem to have run into a distinct problem.

```
Processing 42 classes from /Users/jason/code/compiler-benchmark/compilation/target/scala-2.11/classes with "reflection" generator
Writing out Java source to /Users/jason/code/compiler-benchmark/compilation/target/scala-2.11/src_managed/jmh and resources to /Users/jason/code/compiler-benchmark/compilation/target/scala-2.11/resource_managed/jmh
[info] Compiling 25 Java sources to /Users/jason/code/compiler-benchmark/compilation/target/scala-2.11/classes...
[info] Running scala.bench.ScalacBenchmarkRunner HotScalacBenchmark -foe true -jvm /Library/Java/JavaVirtualMachines/graalvm-0.26/bin/java -jvmArgs -Dgraal.CompilationFailureAction=Silent -psource=scalap -f1
[info] # JMH version: 1.19
[info] # VM version: JDK 1.8.0_121, VM 25.71-b01-internal-jvmci-0.30
[info] # VM invoker: /Library/Java/JavaVirtualMachines/graalvm-0.26/bin/java
[info] # VM options: -Dgraal.CompilationFailureAction=Silent
[info] # Warmup: 10 iterations, 10 s each
[info] # Measurement: 10 iterations, 10 s each
[info] # Timeout: 10 min per iteration
[info] # Threads: 1 thread, will synchronize iterations
[info] # Benchmark mode: Sampling time
[info] # Benchmark: scala.tools.nsc.HotScalacBenchmark.compile
[info] # Parameters: (corpusVersion = a8c43dc, extraArgs = , source = scalap)
[info] 
[info] # Run progress: 0.00% complete, ETA 00:03:20
[info] # Fork: 1 of 1
[info] # Warmup Iteration   1: 14176.748 ms/op
[info] # Warmup Iteration   2: 3959.423 ±(99.9%) 19207.364 ms/op
[info] # Warmup Iteration   3: #
[info] # A fatal error has been detected by the Java Runtime Environment:
[info] #
[info] #  SIGBUS (0xa) at pc=0x000000010eda98c0, pid=41298, tid=0x0000000000005703
[info] #
[info] # JRE version: Java(TM) SE Runtime Environment (8.0_121-b13) (build 1.8.0_121-b13)
[info] # Java VM: Java HotSpot(TM) 64-Bit Graal VM (25.71-b01-internal-jvmci-0.30 mixed mode bsd-amd64 compressed oops)
[info] # Problematic frame:
[info] # J 26225 JVMCI scala.collection.SeqLike$CombinationsItr.next()Ljava/lang/Object; (537 bytes) @ 0x000000010eda98c0 [0x000000010eda9140+0x780] (null)
[info] #
[info] # Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
[info] #
[info] # An error report file with more information is saved as:
[info] # /Users/jason/code/compiler-benchmark/compilation/hs_err_pid41298.log
[info] Loaded disassembler from /Library/Java/JavaVirtualMachines/graalvm-0.26/jre/lib/hsdis-amd64.dylib
[info] #
[info] # If you would like to submit a bug report, please visit:
[info] #   http://bugreport.java.com/bugreport/crash.jsp
[info] #
[info] <forked VM failed with exit code 134>
[info] <stdout last='20 lines'>
[info] #
[info] # A fatal error has been detected by the Java Runtime Environment:
[info] #
[info] #  SIGBUS (0xa) at pc=0x000000010eda98c0, pid=41298, tid=0x0000000000005703
[info] #
[info] # JRE version: Java(TM) SE Runtime Environment (8.0_121-b13) (build 1.8.0_121-b13)
[info] # Java VM: Java HotSpot(TM) 64-Bit Graal VM (25.71-b01-internal-jvmci-0.30 mixed mode bsd-amd64 compressed oops)
[info] # Problematic frame:
[info] # J 26225 JVMCI scala.collection.SeqLike$CombinationsItr.next()Ljava/lang/Object; (537 bytes) @ 0x000000010eda98c0 [0x000000010eda9140+0x780] (null)
[info] #
[info] # Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
[info] #
[info] # An error report file with more information is saved as:
[info] # /Users/jason/code/compiler-benchmark/compilation/hs_err_pid41298.log
[info] Loaded disassembler from /Library/Java/JavaVirtualMachines/graalvm-0.26/jre/lib/hsdis-amd64.dylib
[info] #
[info] # If you would like to submit a bug report, please visit:
[info] #   http://bugreport.java.com/bugreport/crash.jsp
[info] #
[info] </stdout>
[info] <stderr last='20 lines'>
[info] </stderr>
[info] 
[info] Benchmark had encountered error, and fail on error was requested

```

[hs_err_pid41298.log](https://gist.github.com/bf30ad0278723a10ffae8256ac11197d
)

**Comment by tkrodriguez** [comment]
Could you provide some instructions to reproduce this?

**Comment by retronym** [comment]
I've added the instructions to the post above. I haven't been able to reproduce it by just calling the affected method in a loop in a smaller program.

**Comment by tkrodriguez** [comment]
I was able to reproduce the crash it with the latest bits.  I think it's a problem with partial unrolling.

The earlier NPE is definitely the same issue as the commit you linked it.  I'm investigating the crash.

**Comment by tkrodriguez** [comment]
This should be fixed with 8a356b2d96
